### ISSUE TITLE
How do we raise the visibility of the structopt -> clap change?

### ISSUE THREAD

**Issue by epage** [TeXitoi/structopt#525]
I've seen users posting here and elsewhere who are unaware of the migration.  Dependabot, `cargo outdated`, and `cargo upgrade` won't report anything.

**Comment by epage** [comment]
Brainstorming:
- Pin an issue about the maintenance status
- See if rustsec can add this to their warnings discouraging some crates in favor of others
  - The focus here is on "no maintainer" (no one available in case of vulnerability) so it'd be stretching things
- Move this repo into the clap repo
- Add `#[deprecated]` attributes to the trait methods
- Have the derive inject a deprecated function and a call to it

**Comment by TeXitoi** [comment]
* I'm not sure an pinned issue would change anything
* Not sure that's rustsec job, there is no vulerability known in clap v2 for a long time, so no much risk.
* Same, moving the structopt repo will not change.
* Deprecation is annoying.

But maybe the interesting part is that cargo outdated and friend don't report anything. Maybe I can publish a structopt v0.4 (or v1?) that is empty and point to clap v3, for easy discoverability?

**Comment by epage** [comment]
Yeah. didn't call that out but it seems like it'd be nice to have a database for cargo outdated that fills a purpose besides rustsec.

If you are up for a dummy release, that works too.

**Comment by TeXitoi** [comment]
This issue is now pinned, not sure it will change anything, but that's free :-)

**Comment by pinkforest** [comment]
We could flag `informational = unmaintained` on structopt if @TeXitoi agrees ?

This could help people wondering about ansi_term - @epage welcome to send a PR if @TeXitoi is okay with informational advisory.

We flag informational advisories routinely to deprecate or handle superceded crates

People already got `ansi_term` informational and I think this could help people navigate further the issue.

**Comment by BatmanAoD** [comment]
This issue is now 4 years old, and Rustsec now regularly issues "unmaintained" advisories; how do we get one added?

**Comment by BatmanAoD** [comment]
@TeXitoi Are you okay with an "unmaintained" advisory in rustsec? If you'd like, I can clarify in the text that there are no known security issues and that this is really more of a depreciation notice than anything.

**Comment by TeXitoi** [comment]
What would be the advantages? It’s quite some time that clap v3 is out, and the person upgrading their library have switched to clap_derive. Technically, structopt is not "unmaintained", but more an outdated version of clap_derive. It depends (indirectly) on unmaintained crates.

**Comment by BatmanAoD** [comment]
Copying and pasting from [the advisory-db PR](https://github.com/rustsec/advisory-db/pull/2983) in case any followers of this thread are interested in what I think the advantages of an advisory are:

>It's useful for people who still have structopt in a dependency tree, but use something like `cargo deny` to be notified when a crate should be updated or migrated-from. I didn't know that clap has had the same features as structopt for years; I only discovered this thanks to the "unmaintained" advisories for transitive dependencies such as `atty` and went searching to find if `structopt` itself was still maintained.

**Comment by BatmanAoD** [comment]
It's also worth noting that `cargo deny` can be configured to [simply ignore `unmaintained` advisories](https://embarkstudios.github.io/cargo-deny/checks/advisories/cfg.html#the-unmaintained-field-optional), so this is won't cause noise for people who use that tool but are only worried in "actual" vulnerabilities.

I don't know if similar tools, such as `cargo audit`, have any comparable config option.
